sotopia/envs/evaluators.py

```diff
@@ -201,12 +201,34 @@ async def __acall__(
                 structured_output=self.model_name.startswith("custom/structured"),
             )
             response_list = []
-            # Only process evaluations for the actual number of agents
-            for i, evaluation in enumerate(
-                list(response.evaluations.values())[:num_agents]
-            ):
-                # Map agent names to expected format (agent_1, agent_2, etc.)
+
+            # Normalize evaluations to ensure we have entries for all agents
+            evals_dict = dict(response.evaluations)
+            # Preferred keys: agent_1, agent_2, ...
+            expected_keys = [f"agent_{i+1}" for i in range(num_agents)]
+
+            # If model used different keys, try to map deterministically by order
```

Comment:
let's do not do this (from line 210 to line 224) those silent fallback handling is very hard to debug once things are wrong.

let's just match the names we can match and assign things zero if we can not find a good match, also we should log warning

Reply:
also this normalization thing looks a bit weird:

```
class EvaluationForAgents(BaseModel, Generic[T_eval_dim]):
    evaluations: dict[str, T_eval_dim]
```
isn't it a dict already?